processing/src/test/java/org/apache/druid/segment/filter/BaseFilterTest.java

```diff
@@ -816,58 +845,66 @@ private void assertFilterMatches(
+    try {
       Assert.assertEquals(
-          "Cursor (vectorized): " + filter,
+          "Cursor: " + filter,
           expectedRows,
-          selectColumnValuesMatchingFilterUsingVectorCursor(filter, "dim0")
+          selectColumnValuesMatchingFilter(filter, "dim0")
       );
 
+      if (testVectorized) {
+        Assert.assertEquals(
+            "Cursor (vectorized): " + filter,
+            expectedRows,
+            selectColumnValuesMatchingFilterUsingVectorCursor(filter, "dim0")
+        );
+
+        Assert.assertEquals(
+            "Cursor Virtual Column (vectorized): " + filter,
+            expectedRows,
+            selectColumnValuesMatchingFilterUsingVectorVirtualColumnCursor(filter, "vdim0", "dim0")
+        );
+      }
+
       Assert.assertEquals(
-          "Cursor Virtual Column (vectorized): " + filter,
+          "Cursor with postFiltering: " + filter,
           expectedRows,
-          selectColumnValuesMatchingFilterUsingVectorVirtualColumnCursor(filter, "vdim0", "dim0")
+          selectColumnValuesMatchingFilterUsingPostFiltering(filter, "dim0")
       );
-    }
 
-    Assert.assertEquals(
-        "Cursor with postFiltering: " + filter,
-        expectedRows,
-        selectColumnValuesMatchingFilterUsingPostFiltering(filter, "dim0")
-    );
+      if (testVectorized) {
+        Assert.assertEquals(
+            "Cursor with postFiltering (vectorized): " + filter,
+            expectedRows,
+            selectColumnValuesMatchingFilterUsingVectorizedPostFiltering(filter, "dim0")
+        );
+      }
 
-    if (testVectorized) {
       Assert.assertEquals(
-          "Cursor with postFiltering (vectorized): " + filter,
-          expectedRows,
-          selectColumnValuesMatchingFilterUsingVectorizedPostFiltering(filter, "dim0")
+          "Filtered aggregator: " + filter,
+          expectedRows.size(),
+          selectCountUsingFilteredAggregator(filter)
       );
-    }
 
-    Assert.assertEquals(
-        "Filtered aggregator: " + filter,
-        expectedRows.size(),
-        selectCountUsingFilteredAggregator(filter)
-    );
+      if (testVectorized) {
+        Assert.assertEquals(
+            "Filtered aggregator (vectorized): " + filter,
+            expectedRows.size(),
+            selectCountUsingVectorizedFilteredAggregator(filter)
+        );
+      }
 
-    if (testVectorized) {
       Assert.assertEquals(
-          "Filtered aggregator (vectorized): " + filter,
-          expectedRows.size(),
-          selectCountUsingVectorizedFilteredAggregator(filter)
+          "RowBasedColumnSelectorFactory: " + filter,
+          expectedRows,
+          selectColumnValuesMatchingFilterUsingRowBasedColumnSelectorFactory(filter, "dim0")
       );
     }
-
-    Assert.assertEquals(
-        "RowBasedColumnSelectorFactory: " + filter,
-        expectedRows,
-        selectColumnValuesMatchingFilterUsingRowBasedColumnSelectorFactory(filter, "dim0")
-    );
+    catch (ISE ise) {
```

Comment:
Please update the logic to only suppress these `ARRAY<STRING>` arrays when they occur during testing nested columns. Otherwise we might miss `ARRAY<STRING>` related errors that creep in the future in other cases.